### ISSUE TITLE
PDF Unable to be processed by API because of ".pdf does not appear to be a valid PDF"

### ISSUE THREAD

**Issue by jackretterer** [Unstructured-IO/unstructured-api#181]
I passed in a PDF file that is of kind ".pdf" and our api returned an error stating the file wasn't a valid PDF. However, the PDF is able to be processed by Unstructured's python library as a pdf and is identified by our file detection system as a PDF. 

<img width="1175" alt="Screenshot 2023-08-07 at 3 12 04 PM" src="https://github.com/Unstructured-IO/unstructured-api/assets/6834789/0b64e7fc-30dd-4bec-ae6d-86182013706e">

**Comment by cragwolfe** [comment]
You can specify the mimetype for that file this way:

https://github.com/Unstructured-IO/unstructured-api/blob/75eef77/test_general/api/test_app.py#L551

(though agree that this is a bug)

**Comment by yuming-long** [comment]
first inspection looks like a problem with parallel mode, local api doesn't use parallel mode and got the result, so will look into if parallel mode is using the mimetype correctly

**Comment by awalker4** [comment]
Attaching a problematic page. There are 2 layers to the bug:
* The not a valid pdf occurs when we [read an empty page](https://github.com/Unstructured-IO/unstructured-api/blob/f491b85eb3e48fe6d92d7800c4c07d70e88ffa0c/prepline_general/api/general.py#L241) at the start of the call
  * This is actually caused by the parallel mode retry logic. Each page is sent to the api in a BytesIO object. When we get an error response, we send the page a second time, at which point the BytesIO has been read already.
   *  Easy fix - store the page in a buffer before sending. I’ll also add logging to the retry logic so we know when this is happening.
* The real error here is PDFSyntaxError: No /Root object! - Is this really a PDF?
  * We don’t need the api to reproduce:
[bad_page.pdf](https://github.com/Unstructured-IO/unstructured-api/files/12353060/bad_page.pdf)

```
from unstructured.partition.auto import partition

partition(file=open(filename, 'rb'), file_filename="file.pdf", content_type="application/pdf")
```

**Comment by awalker4** [comment]
Minimal example of the No root object error, to match the usage of `extract_pages` over [here](https://github.com/Unstructured-IO/unstructured-inference/blob/4b7276a7e26c6116b4d8e5c6b985f2b16e053734/unstructured_inference/inference/layout.py#L478).

```
from tempfile import NamedTemporaryFile
from pdfminer.high_level import extract_pages

filename = "bad_page.pdf"

# This works!
for page in extract_pages(filename):
    print(page)

# PDFSyntaxError: No /Root object! - Is this really a PDF?
with NamedTemporaryFile() as tmp:
    tmp.write(f.read())
    
    for page in extract_pages(tmp.name):
        print(page)
```

The error goes away when I add a `tmp.flush()`. I think this single page with just an image gets buffered and is not written out fast enough.

**Comment by awalker4** [comment]
Inference fix here - https://github.com/Unstructured-IO/unstructured-inference/pull/178

**Comment by awalker4** [comment]
Second fix is in #186. Once that's merged, I can bump inference to close out this issue.
